activesupport/test/cache/stores/memory_store_test.rb

```diff
@@ -61,6 +63,26 @@ def test_nil_coder_bypasses_mutation_safeguard
     assert_same value, @cache.read("key")
   end
 
+  def test_write_with_unless_exist
+    assert_equal true, @cache.write(1, "aaaaaaaaaa")
+    assert_equal false, @cache.write(1, "aaaaaaaaaa", unless_exist: true)
+    @cache.write(1, nil)
+    assert_equal false, @cache.write(1, "aaaaaaaaaa", unless_exist: true)
+  end
+
+  def test_namespaced_write_with_unless_exist
+    assert_equal true, @namespaced_cache.write(1, "aaaaaaaaaa")
+    assert_equal false, @namespaced_cache.write(1, "aaaaaaaaaa", unless_exist: true)
+    @namespaced_cache.write(1, nil)
+    assert_equal false, @namespaced_cache.write(1, "aaaaaaaaaa", unless_exist: true)
```

Comment:
```suggestion
    @cache = lookup_store(expires_in: 60, namespace: "foo")
    
    assert_equal true, @cache.write(1, "aaaaaaaaaa")
    assert_equal false, @cache.write(1, "aaaaaaaaaa", unless_exist: true)
    @cache.write(1, nil)
    assert_equal false, @cache.write(1, "aaaaaaaaaa", unless_exist: true)
```

Reply:
once I get the other PRs merged, I think this will become more necessary, but 🤷🏻